src/Core/Util.php

```diff
@@ -367,25 +364,18 @@ public static function substr($str, $start = 0, $length = null)
         if (!is_string($str)) {
             throw new InvalidArgumentException('String expected');
         }
-        if (PHP_VERSION_ID < 50400 && $length === null) {
-            $length = self::strlen($str);
-        }
-        if (function_exists('mb_substr')) {
-            // $length calculation above might result in a 0-length string
-            if ($length === 0) {
-                return '';
-            }
-            return mb_substr($str, $start, $length, '8bit');
-        }
+
+        $length = isset($length) ? (int) $length : self::strlen($str);
```

Comment:
We don't need to do this on PHP 5.4+, so the previous checks shouldn't be removed.